doc/tutorials/notifications.md

```diff
@@ -0,0 +1,327 @@
+whether and how they want to receive and handle those notifications
+but not whether notifications are sent.
+ How notifications are sent is
+specified in the *notification configuration* by the system operator.
+
+ Notification configuration
+ ===
+ 
+ System operators can mount the
+desired transport plugins and configure them (e.g. set channel, host,
+port and credentials) either globally or when mounting a configuration
+file.
+ 
+ They need to mount both sending and receiving plugins in order
+to use a transport.
+ 
+```sh
+kdb mount file.ini system/example ini dbussend dbusreceive
+```
+ 
+ How to integrate an I/O-binding and send
+notifications asynchronously
+ ===
+ 
+ Developers do not need to change
+their programs in order to start sending notifications.
+ But without
+the integration of an I/O-binding notifications are sent synchronously
+which will block normal program execution.
+ For programs without time
+constraints (e.g. CLI programs) this may not be important, but for
+GUIs or network services this will have negative impact.
+ 
+ Since many
+different I/O management libraries exist (e.g. libuv, glib or libev)
+the transport plugins use the I/O-interface for their I/O tasks.
+ Each
+I/O management library needs its own I/O-binding.
+ Developers can also
+create their own I/O-binding for the I/O management library of their
+choice.
+ This is described in the last section.
+ 
+ Each I/O-binding
+has its own initialization function that creates a new I/O-binding and
+connects it to the I/O management library.
+ For this tutorial we will
+assume that libuv is used.
+ For details on how to use a specific binding
+please look at the bindings' README.md in `src/bindings/io/<binding>`.
+ 
+```C
+#include <elektra/io_uv.h>
+
+void main (void)
+{
+	KDB* repo;
+
+	// .. open KDB
+
+	// Create libuv event loop
+	uv_loop_t * loop = uv_default_loop ();
+
+	// Initialize I/O binding tied to event loop
+	ElektraIoInterface * binding = elektraIoUvNew (loop);
+
+	// Initialize notification wrapper
+	elektraNotificationOpen (kdb, binding);
+
+	// Start the event loop
+	uv_run (loop, UV_RUN_DEFAULT);
+
+	// Cleanup
+	binding->cleanup (binding);
+	uv_loop_close (loop);
+	free (loop);
+}
+
+void someFunction (void)
+{
+	// notifications are sent asynchronously on kdbSet
+	// over configured transports using our uv event loop
+	kdbSet (repo, key, parentKey);
+}
+```
+ 
+ How to receive notifications
+ ===
+ 
+ We extend the example
+from the previous section where we already created our I/O-binding
+and initialized the notification-wrapper.
+ 
+ In order to handle change
+notifications a developer can either register a variable or a callback.
+Values of registered variables are automatically updated when the value
+of the assigned Key has changed.
+ This is suitable for programs where
+the Key's value is simply displayed or used repeatedly (e.g. by a timer
+or in a loop).
+ When the program's initialization depends on the value
```

Comment:
Structure of text: First explain how variable registering works, then explain where its limits are and how to use callbacks.